django/core/handlers/asgi.py

```diff
@@ -178,38 +178,44 @@ async def handle(self, scope, receive, send):
             body_file = await self.read_body(receive)
         except RequestAborted:
             return
-        # Request is complete and can be served.
-        set_script_prefix(get_script_prefix(scope))
-        await signals.request_started.asend(sender=self.__class__, scope=scope)
-        # Get the request and check for basic issues.
-        request, error_response = self.create_request(scope, body_file)
-        if request is None:
-            body_file.close()
-            await self.send_response(error_response, send)
-            return
-        # Try to catch a disconnect while getting response.
-        tasks = [
-            asyncio.create_task(self.run_get_response(request)),
-            asyncio.create_task(self.listen_for_disconnect(receive)),
-        ]
-        done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
-        done, pending = done.pop(), pending.pop()
-        # Allow views to handle cancellation.
-        pending.cancel()
-        try:
-            await pending
-        except asyncio.CancelledError:
-            # Task re-raised the CancelledError as expected.
-            pass
-        try:
-            response = done.result()
-        except RequestAborted:
-            body_file.close()
-            return
-        except AssertionError:
-            body_file.close()
-            raise
-        # Send the response.
+        with body_file:
```

Comment:
This might be a worthwhile refactoring (or not) but it would be nice to have it as a separate change to not obscure the diff here.